templates/core/terraform/api-webapp/api-webapp.tf

```diff
@@ -0,0 +1,215 @@
+resource "azurerm_app_service_plan" "core" {
+  name                = "plan-${var.resource_name_prefix}-${var.environment}-${var.tre_id}"
+  resource_group_name = var.resource_group_name
+  location            = var.location
+  reserved            = true
+  kind                = "linux"
+  sku {
+    tier     = "PremiumV3"
+    capacity = 1
+    size     = "P1v3"
+  }
+}
+
+resource "azurerm_application_insights" "core" {
+  name                = "appi-${var.resource_name_prefix}-${var.environment}-${var.tre_id}"
+  resource_group_name = var.resource_group_name
+  location            = var.location
+  application_type    = "web"
+}
+
+resource "azurerm_app_service" "management_api" {
+  name                = "api-${var.resource_name_prefix}-${var.environment}-${var.tre_id}"
+  resource_group_name = var.resource_group_name
+  location            = var.location
+  app_service_plan_id = azurerm_app_service_plan.core.id
+
+  identity {
+    type         = "UserAssigned"
+    identity_ids = [azurerm_user_assigned_identity.management_api.id]
+  }
+
+  https_only = true
+  app_settings = {
+
+    "APPINSIGHTS_INSTRUMENTATIONKEY" = azurerm_application_insights.core.instrumentation_key
+  }
+
+  site_config {
+    app_command_line            = "gunicorn -w 2 -k uvicorn.workers.UvicornWorker main:app"
+    remote_debugging_enabled    = false
+    scm_use_main_ip_restriction = true
+    cors {
+      allowed_origins     = []
+      support_credentials = false
+    }
+    always_on       = true
+    min_tls_version = "1.2"
+    ip_restriction {
+      action     = "Deny"
+      ip_address = "0.0.0.0/0"
+      name       = "Deny all"
+      priority   = 2147483647
+    }
+    websockets_enabled = false
+  }
+
+  logs {
+    application_logs {
+      file_system_level = "Information"
+    }
+
+    http_logs {
+      file_system {
+        retention_in_days = 7
+        retention_in_mb   = 100
+      }
+    }
+  }
+}
+
+resource "azurerm_user_assigned_identity" "management_api" {
```

Comment:
This now expects AAD owner isnt it? Can an unpriveleged account create these? Should the identities be created in a different process and we can import them using data block?

Reply:
This could possibly be a system assigned identity too? Reason it was user assigned in this is example is we were using the same identity for multiple apps to simplify assignment of permissions. However think in the architecture we are looking at system assigned will work?

Reply:
I think that it requires only "Managed Identity Contributor" https://docs.microsoft.com/en-us/azure/active-directory/managed-identities-azure-resources/how-to-manage-ua-identity-portal

Reply:
@marrobi good point. we can for now remove this part and rely on system identity, and then add this if we need it later on.